File: Java-CodeServer/src/main/resources/js/prompto/java/JavaThisExpression.js
var JavaExpression = require("./JavaExpression").JavaExpression;
var ThisExpression = require("../expression/ThisExpression").ThisExpression;

function JavaThisExpression() {
    JavaExpression.call(this);
	this.expression = new ThisExpression();
	return this;
}

JavaThisExpression.prototype = Object.create(JavaExpression.prototype);
JavaThisExpression.prototype.constructor = JavaThisExpression;

JavaThisExpression.prototype.toString = function() {
	return this.expression.toString();
};

JavaThisExpression.prototype.toDialect = function(writer) {
    this.expression.toDialect(writer);
};

exports.JavaThisExpression = JavaThisExpression;

File: Java-Core/src/main/java/prompto/java/JavaThisExpression.java
package prompto.java;

import prompto.error.PromptoError;
import prompto.error.SyntaxError;
import prompto.expression.ThisExpression;
import prompto.parser.Section;
import prompto.runtime.Context;
import prompto.type.IType;
import prompto.utils.CodeWriter;

public class JavaThisExpression extends Section implements JavaExpression {
	
	ThisExpression expression = new ThisExpression();
	
	@Override
	public IType check(Context context) throws SyntaxError {
		return expression.check(context);
	}
	
	@Override
	public Object interpret(Context context) throws PromptoError {
		return expression.interpret(context);
	}
	
	@Override
	public void toDialect(CodeWriter writer) {
		expression.toDialect(writer);
	}
}
